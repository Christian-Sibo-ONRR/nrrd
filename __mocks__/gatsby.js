/* eslint-disable no-undef */
const React = require('react')
const gatsby = jest.requireActual('gatsby')

module.exports = {
  ...gatsby,
  graphql: ([queryString]) => getQueryName(queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) =>
    React.createElement('a', {
      ...rest,
      href: to,
      'data-testId': 'BaseLink'
    })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: function (queryName) {
    const filePath = `./__mock_queries__/${ queryName }_results.json`

    try {
      return require(filePath)
    }
    catch (err) {
      console.warn(`Failed to require mock file for ${ queryName }`)
      return {}
    }
  },
}
function getQueryName (queryString) {
  const captureGroup = queryString.match(/query\s([^\s]+).*{/)
  if (captureGroup && captureGroup.length) {
    return captureGroup[1]
  }
}
