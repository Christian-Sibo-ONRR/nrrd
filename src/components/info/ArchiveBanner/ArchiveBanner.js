import React from 'react'

import makeStyles from '@material-ui/core/styles/makeStyles'
import useTheme from '@material-ui/styles/useTheme'

import GlossaryTerm from '../../GlossaryTerm/GlossaryTerm'

import { IconArchiveImg } from '../../images'

// TODO: update styles, should be able to utilize main theme
const useStyles = makeStyles(theme => ({
  root: {
    backgroundColor: '--grayPale',
    borderLeft: '10px solid gray',
    padding: '1.125em .5em .5em .5em',
    marginBottom: '2rem',
  },
  content: {
    margin: '0 0 1rem 0',
    color: '--blackLight',
    fontSize: '0.8125rem',
    fontWeight: '300',
    lineHeight: '1.1875rem',
    marginBottom: '0.625rem',
    padding: '0px',
    '> svg': {
      verticalAlign: 'bottom',
    }
  }
}))

const ArchiveBanner = () => {
  const theme = useTheme()
  const classes = useStyles(theme)

  return (
    <div className={classes.root}>
      <p className={classes.content}>
        <IconArchiveImg /> This content was created as part of <GlossaryTerm>USEITI</GlossaryTerm> and is no longer being updated.
      </p>
    </div>
  )
}

export default ArchiveBanner
