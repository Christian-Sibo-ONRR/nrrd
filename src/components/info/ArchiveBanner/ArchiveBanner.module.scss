@value colors: "../../../css-global/colors.scss";
@value grayPale as --grayPale, blackLight as --blackLight from colors;

.root {
	background-color: --grayPale;
  border-left: 10px solid gray;
  padding: 1.125em .5em .5em .5em;
  margin-bottom: 2rem;
}

.content {
  margin: 0 0 1rem 0;
  color: --blackLight;
  font-size: 0.8125rem;
  font-weight: 300;
  line-height: 1.1875rem;
  margin-bottom: 0.625rem;
  padding: 0px;

  > svg {
    vertical-align: bottom;
  }
}