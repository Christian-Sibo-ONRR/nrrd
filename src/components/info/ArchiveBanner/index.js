export { default as ArchiveBanner } from './ArchiveBanner'
