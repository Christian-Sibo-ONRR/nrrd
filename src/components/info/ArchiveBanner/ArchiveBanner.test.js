/* eslint-disable no-undef */
import React from 'react'
import ArchiveBanner from './ArchiveBanner'
import renderer from 'react-test-renderer'
import theme from '../../../js/mui/theme'
import { ThemeProvider } from '@material-ui/core/styles'
import { withPrefix } from 'gatsby'

describe('Archive Banner component:', () => {
  function MockedTheme ({ children }) {
    return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    )
  }

  it('Render success', () => {
    const tree = renderer
      .create(<MockedTheme><ArchiveBanner /></MockedTheme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
