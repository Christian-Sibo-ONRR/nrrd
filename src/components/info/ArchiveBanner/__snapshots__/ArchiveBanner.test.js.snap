// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive Banner component: Render success 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="makeStyles-content-2"
  >
    <test-file-stub
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
    />
     This content was created as part of 
    <span
      aria-describedby={null}
      className="makeStyles-term-5"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      title="The Extractive Industries Transparency Initiative Standard is an international standard for openness around the management of revenue from natural resources. Governments disclose how much they receive from extractive companies operating in their country and these companies disclose how much they pay. Governments sign up to implement the EITI Standard and must meet seven requirements. In 2017, the U.S. withdrew from EITI as an Implementing Country, but remains committed to institutionalizing the EITI principles of transparency and accountability."
    >
      USEITI 
      <test-file-stub
        style={
          Object {
            "fill": "currentColor",
            "height": "16px",
            "marginLeft": "1.6px",
            "width": "16px",
          }
        }
      />
    </span>
     and is no longer being updated .
  </p>
</div>
`;
